lib/phoenix/controller.ex

```diff
@@ -627,8 +629,11 @@ defmodule Phoenix.Controller do
 
     view = Map.get(conn.private, :phoenix_view) ||
             raise "a view module was not specified, set one with put_view/2"
-    data = Phoenix.View.render_to_iodata(view, template,
-                                         Map.put(conn.assigns, :conn, conn))
+
+    data = instrument conn, :phoenix_controller_render, template <> format, fn ->
```

Comment:
Let's make the runtime a map and enforce it. This will help us avoid running into exactly the mistake we did here. What if we want to add more information in the future? Because we are passing `template <> format`, it would be a breaking change. If we use a map, you can always add new keys.